common-api/src/main/java/org/pmiops/workbench/config/CacheSpringConfiguration.java

```diff
@@ -66,14 +70,30 @@ public static FeaturedWorkspacesConfig lookupFeaturedWorkspacesConfig(
     return (FeaturedWorkspacesConfig) configCache.get(DbConfig.FEATURED_WORKSPACES_CONFIG_ID);
   }
 
-  @Bean
+  /**
+   * Request-scoped WorkbenchConfig bean. This version should be used in all cases where updates
+   * during execution are desired. It may not be used in an Autowired Constructor or Bean function
+   */
+  @Bean(name = WORKBENCH_CONFIG_REQUEST_SCOPED)
+  @Primary
   @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
   WorkbenchConfig getWorkbenchConfig(
       @Qualifier("configCache") LoadingCache<String, Object> configCache)
       throws ExecutionException {
     return lookupWorkbenchConfig(configCache);
   }
 
+  /**
+   * Singleton version for use in Service constructors and Bean function arguments. Not updated
+   * dynamically
+   */
+  @Bean(name = WORKBENCH_CONFIG_SINGLETON)
```

Comment:
Not sure how often workbench config changes happen off-schedule or without an app deploy. If this does happen off-schedule, seems like this could bite us.

Reply:
The primary bean is still the request-scoped one. How do you mean "bite"?

Reply:
Is there a chance that MonitoredResource could be used places besides a constructor and gets stale if the workbench config changes without a restart/deploy?

Reply:
Yes, but that's by design. There are things in the config that are dynamically scoped but that should never be changed at runtime. for instance the project names, environment name, etc. (I'm working on a config document now, because I'm seeing both low-hanging fruit and issues on the horizon. The one-size-fits-all nature of how we provide it is one issue.)

I intend the MonitoredResource to be created once per process, since its job is to describe the "resource being monitored", i.e. this node of this project, with this namespace and location. It has two purposes: (1) to allow multiple processes to write to the same time series without trigging rate limit exceptions, and (2) to provide context to all the graphs for the thing being monitored.

The "scope" of it should be the entire lifetime of the process. Since we can't change regions or projectIds during execution, it would be an error to pick them up here.

But, and here's the sad thing, if I did move all that to a request scope, I wouldn't be able to use it in constructors or when building singleton-scoped things. But it's effectively a constant for the lifetime of the process.

Reply:
Thanks for the info, that helps a lot.